tests/test_default_response_class.py

```diff
@@ -31,6 +32,12 @@ def get_root():
     return {"msg": "Hello World"}
 
 
+@app.get("/orjson_non_str_keys")
+def get_orjson_non_str_keys():
+    key = quoted_name("msg")
+    return {key: "Hello World"}
```

Comment:
Isn't that `key` still a string? It seems like it should be working even before the fix (I didn't check that though).

Also, you could provide a test with some integer as a key maybe?